openpmd_validator/check_h5.py

```diff
@@ -392,7 +397,7 @@ def check_root_attr(f, v):
     result_array += test_attr(f, v, "required", "iterationFormat", np.string_)
 
     #   optional but required for extensions
-    result_array += test_attr(f, v, "optional", "openPMDextension", np.string_, "^[a-zA-Z0-9-;]+$")
+    result_array += test_attr(f, v, "optional", "openPMDextension", np.string_, "^[a-zA-Z0-9\-;]+$")
```

Comment:
this is a bug-fix for extension names from the last PR. The `-` needs an escape, otherwise it's a range. (`\` is not allowed, would also need an escape.)

Reply:
Since regex can sometimes be cryptic, would you add a comment explaining what it does? Like
```
# Check that openPMDextension contains only letters, or digits, or the characters `-` and `;`
```

Reply:
yes, great point!